index.js

```diff
@@ -354,4 +358,38 @@ formMobile.addEventListener('submit', (event) => {
   } else {
     messageMob.textContent = '';
   }
-});
\ No newline at end of file
+});
+
+// ...portfolio-Preserve data in the browser...
+
+formMobile.addEventListener('change', () => {
+  const formDataMobile = {
+    name: mobileName.value,
+    email: emailMob.value,
+    text: mobileText.value,
+  };
+  const stringifyFormDataMobile = JSON.stringify(formDataMobile);
+  localStorage.setItem('mobileData', stringifyFormDataMobile);
+});
+
+formDesktop.addEventListener('change', () => {
+  const formDataDesktop = {
+    name: desktopName.value,
+    email: email.value,
+    text: desktopText.value,
+  };
+  const stringifyFormDataDesktop = JSON.stringify(formDataDesktop);
+  localStorage.setItem('desktopData', stringifyFormDataDesktop);
+});
+
+const storedMobileData = JSON.parse(localStorage.getItem('mobileData'));
+const storedDesktopData = JSON.parse(localStorage.getItem('desktopData'));
+
+window.addEventListener('load', () => {
+  mobileName.value = storedMobileData.name;
+  emailMob.value = storedMobileData.email;
+  mobileText.value = storedMobileData.text;
+  desktopName.value = storedDesktopData.name;
+  email.value = storedDesktopData.email;
+  desktopText.value = storedDesktopData.text;
+});
```

Comment:
- [ ] [OPTIONAL] Great job making the changes but it would be better if you check whether the local storage is empty or not before setting the input values as it throws an error in the console when accessing a null value.
  ![image](https://user-images.githubusercontent.com/104968236/232046547-86f2a804-b442-49f8-a8a5-0f464cd4c746.png)
- You can add something like this:
```
// Check if the form data object is found on localStorage
if (storedMobileData) {
  mobileName.value = storedMobileData.name;
  emailMob.value = storedMobileData.email;
  mobileText.value = storedMobileData.text;
}
if (storedDesktopData) {
  desktopName.value = storedDesktopData.name;
  email.value = storedDesktopData.email;
  desktopText.value = storedDesktopData.text;
}

```
- Cheers and happy coding 🎉

Reply:
Thank you.